util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,28 @@
+package com.twitter.util
+
+import java.util.logging.{Level, Logger}
+
+import scala.collection.mutable
+
+class Batcher[In, Out](
+  sizeThreshold: Int,
+  timeThreshold: Duration = Duration.Top,
+  sizePercentile: => Float = 1.0f,
+  f: Seq[In] => Future[Seq[Out]]
+)(
+  implicit timer: Timer
+) extends Function1[In, Future[Out]] { batcher =>
+  import java.util.logging.Level.WARNING
+
+  val executor = new BatchExecutor[In, Out](sizeThreshold, timeThreshold, sizePercentile, f)
+
+  def apply(t: In): Future[Out] = executor.enqueue(t)
+
+  def flushBatch() = {
+    val doAfter = synchronized {
```

Comment:
I wrote some more tests to verify these scenarios and make sure that regular execution, manual flushes, and timed flushes don't interfere with each other.  Please let me know if you think the test coverage is sufficient.